Week3/homework/4-step3.js

```diff
@@ -1,2 +1,20 @@
-'use strict';
-// paste your freeCodeCamp solutions in here
+// 'use strict';
+// // paste your freeCodeCamp solutions in here
+
+// function Dog() {
+//     this.name = "Dante";
+//     this.color = "black";
+//     this.numLegs = 4;
+// }
+
+// // second part
+
+// function Dog() {
+//     this.name = "Rupert";
+//     this.color = "brown";
+//     this.numLegs = 4;
+//   }
+//   // Add your code below this line
+  
+//   let hound = new Dog();
```

Comment:
Hi @SignosRk - what if I want to have 10 dogs. I would have to create 10 different functions in that case. In such a scenario, it is a good idea to provide arguments to your constructor function and then use that to create all objects. For example:

```
function Dog(name, color, legs) {
  this.name = name;
  this.color = color;
  this.legs = legs;
}

const dog1 = new Dog('Dante', 'black', 4);
const dog2 = new Dog('Rupert', 'brown', 4);
```

Makes sense?

Reply:
of course it makes sense, it would be the proper thing to do if you have more dogs! I just didn't think to add a function and pass those arguments.. Good thing man...   :+1: